packages/beacon-node/src/chain/archiver/archiveStates.ts

```diff
@@ -43,22 +47,23 @@ export class StatesArchiver {
   async maybeArchiveState(finalized: CheckpointWithHex): Promise<void> {
     const lastStoredSlot = await this.db.stateArchive.lastKey();
     const lastStoredEpoch = computeEpochAtSlot(lastStoredSlot ?? 0);
+    const archiveStateEpochFrequency = this.opts.archiveStateEpochFrequency;
 
-    if (finalized.epoch - lastStoredEpoch > PERSIST_TEMP_STATE_EVERY_EPOCHS) {
+    if (finalized.epoch - lastStoredEpoch > Math.min(PERSIST_TEMP_STATE_EVERY_EPOCHS, archiveStateEpochFrequency)) {
```

Comment:
why is this line changed? seems unrelated to the PR

Reply:
Added some words to the PR description